File: GMTool/app/view/game/alluserInfo.py
from django.shortcuts import render
from django.http import HttpResponse
from app.infra.db import *
from datetime import datetime
import json

def inquiryUserInfo(request):
	if request.method == 'GET':
		return render(
			request,
			'app/game/inquiryUserInfo.html'
		)

	else:
		response_data = request.body.decode('utf-8')
		response_data = json.loads(response_data)

		msg = response_data['msg']
		settings = response_data['settings']
		datas = {}
		if msg == 'getUserInfo':
			datas = getUserInfo(settings)
			datas = json.dumps(datas, ensure_ascii=False).encode('utf-8')

		elif msg == 'updateUserInfo':
			updateList = response_data['updateList']
			updateUserInfo(settings, updateList)

		elif msg == 'deleteUserInfo':
			deleteList = response_data['deleteList']
			deleteUserInfo(settings, deleteList)

		return HttpResponse(datas, content_type='application/json; charset=utf-8')


def getUserInfo(settings):
	userInfoType = settings['userInfoType']
	rows = {}

	with dbConnector(runMode=settings['runMode'], dbname='knightrun') as commander:
		userId = getUserId(commander, settings)

		if userInfoType == 'accounts':
			rows = commander.execute("SELECT userId, nickName FROM accounts WHERE userId = %s", str(userId))

		elif userInfoType == 'users':
			rows = commander.execute("SELECT gold , gem FROM users WHERE userId = %s", str(userId))

		elif userInfoType == 'units':
			rows = commander.execute("SELECT a.id, b.unitName, a.level, a.count, a.rankPoint, a.rankLevel FROM units AS a JOIN _common_unitName AS b ON a.id = b.unitId WHERE userId = %s", str(userId))

		elif userInfoType == 'cards':
			rows = commander.execute("SELECT a.id, b.cardName, a.level, a.count FROM cards AS a JOIN _common_cardName AS b ON a.id = b.cardId WHERE userId = %s", str(userId))

		elif userInfoType == 'stage_levels':
			rows = commander.execute("SELECT stageLevelId, clearCount FROM stage_levels WHERE userId = %s", str(userId))

		elif userInfoType == 'armors':
			rows = commander.execute("SELECT a.id, b.Name, a.level, a.count, a.unitId FROM armors AS a JOIN _common_equipName AS b ON a.id = b.Id WHERE userId =%s", str(userId))

		elif userInfoType == 'weapons':
			rows = commander.execute("SELECT a.id, b.Name, a.level, a.count, a.unitId FROM weapons AS a JOIN _common_equipName AS b ON a.id = b.Id WHERE userId =%s", str(userId))

		elif userInfoType == 'decks':
			rows = commander.execute("SELECT modeType, deckNum, orderNum, slotType, classId, deckName FROM decks WHERE userId = %s", str(userId))

		elif userInfoType == 'energy':
			rows = commander.execute("SELECT type, id, boxEnergy, userEnergy, userBonusEnergy FROM infusion_boxes  WHERE userID = %s", str(userId))

		return rows

def updateUserInfo(settings, datas):
	userInfoType = settings['userInfoType']

	with dbConnector(runMode=settings['runMode'], dbname='knightrun', auto_trans=True) as commander:
		userId = getUserId(commander, settings)

		for data in datas:
			if userInfoType == 'accounts':
				commander.execute("UPDATE accounts SET nickName = %s WHERE userId = %s", str(data['nickName']), str(userId))

			elif userInfoType == 'users':
				commander.execute("UPDATE users SET gold = %s , gem = %s WHERE userId = %s", str(data['gold']), str(data['gem']), str(userId))

			elif userInfoType == 'units':
				commander.execute("UPDATE units SET level = %s, count = %s, rankPoint = %s, rankLevel = %s WHERE userId = %s AND id = %s", str(data['level']), str(data['count']), str(data['rankPoint']), str(data['rankLevel']), str(userId), str(data['id']))

			elif userInfoType == 'cards':
				commander.execute("UPDATE cards SET level = %s, count = %s WHERE userId = %s AND id = %s", str(data['level']), str(data['count']), str(userId), str(data['id']))

			elif userInfoType == 'armors':
				commander.execute("UPDATE armors SET level = %s, count = %s, unitId = %s WHERE userId = %s AND id = %s", str(data['level']), str(data['count']), str(data['unitId']), str(userId), str(data['id']))

			elif userInfoType == 'weapons':
				commander.execute("UPDATE weapons SET level = %s, count = %s, unitId = %s WHERE userId = %s AND id = %s", str(data['level']), str(data['count']), str(data['unitId']), str(userId), str(data['id']))

			elif userInfoType == 'stage_levels':
				commander.execute("UPDATE stage_levels SET clearCount = %s WHERE userId = %s AND stageLevelId = %s", str(data['clearCount']), str(userId), str(data['stageLevelId']))

			elif userInfoType == 'decks':
				query = "UPDATE decks SET classId = {0}, deckName = '{1}' WHERE userId = {2} AND modeType = {3} AND deckNum = {4} AND slotType = {5} AND orderNum = {6}"\
							.format(data['classId'], data['deckName'], userId, data['modeType'], data['deckNum'], data['slotType'], data['orderNum'])

				commander.execute(query)

			elif userInfoType == 'energy':
				 commander.execute("UPDATE infusion_boxes SET id = %s, boxEnergy = %s, userEnergy =%s, userBonusEnergy =%s WHERE userId = %s AND type =%s" , str(data['id']) , str(data['boxEnergy']) , str(data['userEnergy']), str(data['userBonusEnergy']),  str(userId), str(data['type']) )

def deleteUserInfo(settings, datas):
	userInfoType = settings['userInfoType']

	with dbConnector(runMode=settings['runMode'], dbname='knightrun', auto_trans=True) as commander:
		userId = getUserId(commander, settings)

		for data in datas:
			if userInfoType == 'accounts':
				args = (
					userId,
				)
				commander.callproc('p_deleteUserDatas', args)

			elif userInfoType == 'users':
				commander.execute("DELETE FROM users WHERE userId = %s", str(userId))

			elif userInfoType == 'units':
				commander.execute("DELETE FROM units WHERE userId = %s AND id = %s", str(userId), str(data['id']))

			elif userInfoType == 'cards':
				commander.execute("DELETE FROM cards WHERE userId = %s AND id = %s", str(userId), str(data['id']))

			elif userInfoType == 'armors':
				commander.execute("DELETE FROM armors WHERE userId = %s AND id = %s", str(userId), str(data['id']))

			elif userInfoType == 'weapons':
				commander.execute("DELETE FROM weapons WHERE userId = %s AND id = %s", str(userId), str(data['id']))

			elif userInfoType == 'stage_levels':
				commander.execute("DELETE FROM stage_levels WHERE userId = %s AND stageLevelId = %s", str(userId), str(data['stageLevelId']))

			elif userInfoType == 'decks':
				commander.execute("DELETE FROM decks WHERE userId = %s AND modeType = %s AND deckNum = %s AND orderNum = %s AND slotType = %s AND classId = %s",
					str(userId), str(data['modeType']), str(data['deckNum']), str(data['orderNum']), str(data['slotType']), str(data['classId'])
				)
			elif userInfoType == 'energy':
				commander.execute("DELETE FROM infusion_boxes WHERE userId = %s AND type = %s AND id = %s", str(userId), str(data['type']), str(data['id']))

def getUserId(commander, settings):
	
	#identifierText = {"userID":0}
	identifierText = 0

	return identifierText

def allUserModify(request):
	if request.method == 'GET':
		return render(
			request,
			'app/game/allUserModify.html'
		)

	else:
		response_data = request.body.decode('utf-8')
		response_data = json.loads(response_data)

		data = response_data['updateMap']
		settings = response_data['settings']
		userInfoType = settings['userInfoType']

		with dbConnector(runMode=settings['runMode'], dbname='knightrun', auto_trans=True) as commander:
			userId = getUserId(commander, settings)
			
			if userInfoType == 'accounts':
				pass

			elif userInfoType == 'users':
				#commander.execute("UPDATE users SET gold = %s, gem = %s WHERE userId = 224", str(data['gold']), str(data['gem']))
				commander.execute("UPDATE users SET gold = %s ", str(data['gold']))

			elif userInfoType == 'units':
				#commander.execute("INSERT INTO units(userId, id, level, count, rankPoint, rankLevel) VALUES(%s, %s, %s, %s, %s, %s)", str(userId), str(data['id']), str(data['level']), str(data['count']), str(data['rankPoint']), str(data['rankLevel']))
				commander.execute("UPDATE units SET level =%s , count=%s , rankPoint= %s , rankLevel=%s WHERE id = %s ",  str(data['level']), str(data['count']), str(data['rankPoint']), str(data['rankLevel']), str(data['id']))

			elif userInfoType == 'cards':
				#commander.execute("INSERT INTO cards(userId, id, level, count) VALUES(%s, %s, %s, %s)", str(userId), str(data['id']), str(data['level']), str(data['count']))
				commander.execute("UPDATE cards SET level = %s, count=%s WHERE id = %s", str(data['level']), str(data['count']), str(data['id']))

			elif userInfoType == 'armors':
				#commander.execute("INSERT INTO armors(userId, id, level, count, unitId) VALUES(%s, %s, %s, %s, %s)", str(userId), str(data['id']), str(data['level']), str(data['count']), str(data['unitId']))
				commander.execute("UPDATE armors SET level = %s, count=%s WHERE id = %s", str(data['level']), str(data['count']), str(data['id']))

			elif userInfoType == 'weapons':
				#commander.execute("INSERT INTO weapons(userId, id, level, count, unitId) VALUES(%s, %s, %s, %s, %s)", str(userId), str(data['id']), str(data['level']), str(data['count']), str(data['unitId']))
				commander.execute("UPDATE weapons SET level =%s, count=%s WHERE id = %s",  str(data['level']), str(data['count']), str(data['id']))
			elif userInfoType == 'allUnit':
				units = [1001,1002,1003,1004,1005,1006,1007,1008,1009,1010,1011,1012]
				userIds = commander.execute("select distinct(userid) from units;")
				for userId in userIds:
					for unit in units:
						commander.execute("Insert IGNORE  into units(userId, id, level ) values(%s, %s, %s)", userId['userid'], unit, str(data['level']))
			elif userInfoType == 'allSkill':
				skills = [101,102,103,104,105,201,202,203,204,205,301,302,303,304,305,401,402,403,404,405,501,502,503,504,505,601,602,603,604,605,701,702,703,704,705,801,802,803,804,805,901,902,903,904,905,1001,1002,1003,1004,1005,1101,1102,1103,1104,1105,1201,1202,1203,1204,1205]
				userIds = commander.execute("select distinct(userid) from cards;")
				for userId in userIds:
					for skill in skills:
						commander.execute("Insert IGNORE  into cards(userId, id,level ) values(%s, %s, %s)", userId['userid'], skill, str(data['level']))
			elif userInfoType == 'allWeapons':
				weapons = [1001,1002,1003,1004,1005,1006,1007,1008,1009,1010,1011]
				userIds = commander.execute("select distinct(userid) from cards;")
				for userId in userIds:
					for skill in weapons:
						commander.execute("Insert IGNORE  into weapons(userId, id, level ) values(%s, %s, %s)", userId['userid'], skill, str(data['level']))

			elif userInfoType == 'allUnitsLevel':
				commander.execute("update units set level = %s ;",str(data['level']) )
			
			elif userInfoType == 'allSkillsLevel':
				commander.execute("update cards set level = %s ;",str(data['level']) )
			
			elif userInfoType == 'allWeaponsLevel':
				commander.execute("update weapons set level = %s ;",str(data['level']) )
				





		return HttpResponse({}, content_type='application/json; charset=utf-8')

File: GMTool/app/static/gmtool/controller/alluserInfo.js
krApp.controller('userInfoController', function ($scope, $http) {
    $scope.runMode = '';
    $scope.userInfoType = '';
    $scope.identifierText = '';
    $scope.header_names = [];
    $scope.datas = [];
    $scope.isAllChoiceCheck = false;

    $scope.normal_key_map = {
        'accounts': ['userId'],
        'users':['userId'],
        'units': ['id','unitName'],
        'cards': ['id','cardName'],
        'armors': ['id','Name'],
        'weapons': ['id'],
        'stage_levels': ['stageLevelId'],
        'decks': ['modeType', 'deckNum', 'slotType', 'orderNum']
    }

    $scope.updateUserInfoSettings = function () {
        $scope.runMode = $('#runMode').val();
        $scope.userInfoType = $('#userInfoType').val();
        $scope.identifierType = $('#identifierType').val();
        $scope.identifierText = $('#identifierText').val();
    }

    $scope.getUserInfoSettings = function () {
        return {
            "runMode": $scope.runMode,
            "userInfoType": $scope.userInfoType,
            "identifierType": $scope.identifierType,
            "identifierText": $scope.identifierText
        };
    }

    $scope.allUserInfo = function () {
        $scope.updateUserInfoSettings();
      
        if ($scope.userInfoType == '' ) return;

        let data_map = {
            'settings': $scope.getUserInfoSettings(),
            'updateMap': {}
        }

        if ($scope.userInfoType == 'accounts') {

        }
        else if ($scope.userInfoType == 'users') {
            let gold = $('#users_gold').val();
            //let gem = $('#users_gem').val();

            data_map['updateMap'] = {
                'gold': gold,
            //    'gem': gem
            };

        }
        else if ($scope.userInfoType == 'cards') {
            let id = $('#cards_cardId').val();
            let level = $('#cards_level').val();
            let count = $('#cards_count').val();

            data_map['updateMap'] = {
                'id': id,
                'level': level,
                'count': count
            };
        }
        else if ($scope.userInfoType == 'units') {
            let id = $('#units_unitId').val();
            let level = $('#units_level').val();
            let count = $('#units_count').val();
            let rankPoint = $('#units_rankPoint').val();
            let rankLevel = $('#units_rankLevel').val();

            data_map['updateMap'] = {
                'id': id,
                'level': level,
                'count': count,
                'rankPoint': rankPoint,
                'rankLevel': rankLevel
            };
        }
        else if ($scope.userInfoType == 'armors') {
            let id = $('#armors_id').val();
            let level = $('#armors_level').val();
            let count = $('#armors_count').val();
            let unitId = $('#armors_unitId').val();

            data_map['updateMap'] = {
                'id': id,
                'level': level,
                'count': count,
                'unitId': unitId
            };
        }
        else if ($scope.userInfoType == 'weapons') {
            let id = $('#weapons_id').val();
            let level = $('#weapons_level').val();
            let count = $('#weapons_count').val();
            let unitId = $('#weapons_unitId').val();

            data_map['updateMap'] = {
                'id': id,
                'level': level,
                'count': count,
                'unitId': unitId
            };
        }
        else if ($scope.userInfoType == 'stage_levels') {
            let stageLevelId = $('#stage_levels_stageLevelId').val();
            let clearCount = $('#stage_levels_clearCount').val();

            data_map['updateMap'] = {
                'stageLevelId': stageLevelId,
                'clearCount': clearCount
            };
        }
        else if ($scope.userInfoType == 'decks') {
            let modeType = $('#decks_modeType').val();
            let deckNum = $('#decks_deckNum').val();
            let orderNum = $('#decks_orderNum').val();
            let slotType = $('#decks_slotType').val();
            let classId = $('#decks_classId').val();
            let deckName = $('#decks_deckName').val();

            data_map['updateMap'] = {
                'modeType': modeType,
                'deckNum': deckNum,
                'orderNum': orderNum,
                'slotType': slotType,
                'classId': classId,
                'deckName': deckName
            }
        }
        else if ($scope.userInfoType == 'allUnit') {
            
            let level = $('#alulevel').val();


            data_map['updateMap'] = {
                'level': level
            }
        }
        else if ($scope.userInfoType == 'allSkill') {
            let level = $('#alslevel').val();

            data_map['updateMap'] = {
                'level': level
            }
        }
        else if ($scope.userInfoType == 'allWeapons') {
            let level = $('#alwlevel').val();

            data_map['updateMap'] = {
                'level': level
            }
        }
        else if ($scope.userInfoType == 'allUnitsLevel') {
            let level = $('#aulevel').val();

            data_map['updateMap'] = {
                'level': level
            }
        }
        else if ($scope.userInfoType == 'allSkillsLevel') {
            let level = $('#aslevel').val();

            data_map['updateMap'] = {
                'level': level
            }
        }
        else if ($scope.userInfoType == 'allWeaponsLevel') {
            let level = $('#awlevel').val();

            data_map['updateMap'] = {
                'level': level
            }
        }

        $scope.post('/game/allUserModify/', data_map, function (data) {
            alert('succeed to add user infomation');
        }, function (res) {
            alert('failed to add user infomation, {0}, {1}'.format(res.status, res.statusText));
        });
    }

    $scope.onUserInfoTypeChange = function (selectedUserInfoType) {
        $scope.userInfoType = selectedUserInfoType;
    }

    $scope.getUserInfo = function () {
        $scope.updateUserInfoSettings();

        if ($scope.userInfoType == '' || $scope.identifierText == '') return;

        let data_map = {
            'msg': 'getUserInfo',
            'settings': $scope.getUserInfoSettings()
        }

        $scope.post('/game/inquiryUserInfo/', data_map, function (data) {
            $scope.datas = data;
            $scope.header_names = Object.keys($scope.datas[0])

            $scope.makeElementHtmlOfUserInfo($scope.datas)
        }, function (res) {
            alert('failed to inquiry user infomation, {0}, {1}'.format(res.status, res.statusText));
        });
    }

    $scope.makeElementHtmlOfUserInfo = function (datas) {
        let tbody = document.getElementById('tbodyUserInfo');
        tbody.innerHTML = '';
        
        for (let tr_count = 0; tr_count < datas.length; tr_count++) {
            let data = datas[tr_count];

            let tr = tbody.insertRow(tr_count);
            tr.id = 'userInfoElement_{0}'.format(tr_count)

            let td_count = 0;
            html = "<input type='checkbox' class='form-control' name='isChecked[]' value='{0}' />".format(tr_count);
            var cell = tr.insertCell(td_count++);
            cell.innerHTML = html;

            for (let key in data) {
                let value = data[key];

                var cell = tr.insertCell(td_count++);
                if ($scope.userInfoType in $scope.normal_key_map && $scope.normal_key_map[$scope.userInfoType].indexOf(key) >= 0) {
                    cell.innerHTML = value;
                }
                else {               
                    html = "<input type='{0}' class='form-control' value='{1}' style='float:left; margin:auto' />&nbsp;\n";    
                    if (typeof (value) == "string") {
                        html = html.format("text", value);

                    }
                    else if (typeof (value) == "number") {
                        html = html.format("number", value);
                    }
                    
                    cell.innerHTML = html;
                }
            }
        }
    }

    $scope.post = function (url, data, success_callback, fail_callback) {
        let req = {
            method: 'POST',
            url: url,
            headers: {
                'Accept': 'application/json; charset=utf-8',
                'Content-Type': 'application/json; charset=utf-8'
            },
            data: data,
            dataType: 'json'
        };

        $http(req).then(function (response) {
            success_callback(response.data);

        }, function (response) {
            fail_callback(response);
        });
    }
});